vnext/Shared/InspectorPackagerConnection.cpp

```diff
@@ -0,0 +1,176 @@
+
+#include "pch.h"
+
+#include <folly/json.h>
+#include "InspectorPackagerConnection.h"
+
+namespace Microsoft::ReactNative {
+
+namespace {
+
+struct InspectorProtocol {
+  static constexpr const char InspectorMessage_EVENT[] = "event";
+  static constexpr const char InspectorMessage_PAGEID[] = "pageId";
+  static constexpr const char InspectorMessage_PAYLOAD[] = "payload";
+
+  static constexpr const char InspectorMessage_eventName_wrappedEvent[] = "wrappedEvent";
+  static constexpr const char InspectorMessage_eventName_getPages[] = "getPages";
+  static constexpr const char InspectorMessage_eventName_connect[] = "connect";
+  static constexpr const char InspectorMessage_eventName_disconnect[] = "disconnect";
+
+  enum class EventType { GetPages, WrappedEvent, Connect, Disconnect };
+
+  static EventType getEventType(const folly::dynamic &messageFromPackager) {
+    std::string event = messageFromPackager.at(InspectorProtocol::InspectorMessage_EVENT).getString();
+    if (!event.compare(InspectorMessage_eventName_getPages))
+      return EventType::GetPages;
+    if (!event.compare(InspectorMessage_eventName_wrappedEvent))
+      return EventType::WrappedEvent;
+    if (!event.compare(InspectorMessage_eventName_connect))
+      return EventType::Connect;
+    if (!event.compare(InspectorMessage_eventName_disconnect))
+      return EventType::Disconnect;
```

Comment:
consider comparing against == 0 so as to not confuse people into thinking this means "if they compare as not-equal" :)

Reply:
I'll incorporate this in the other PR.